noir-projects/noir-contracts/contracts/app/token_contract/src/main.nr

```diff
@@ -488,6 +488,40 @@ pub contract Token {
     }
     // docs:end:finalize_transfer_to_private
 
+    #[private]
+    fn finalize_transfer_to_private_from_private(
+        from: AztecAddress,
+        partial_note: PartialUintNote,
+        amount: u128,
+        nonce: Field,
+    ) {
+        if (!from.eq(context.msg_sender())) {
+            assert_current_call_valid_authwit(&mut context, from);
+        } else {
+            assert(nonce == 0, "invalid nonce");
+        }
+
+        // First we subtract the `amount` from the private balance of `from`
+        storage.balances.at(from).sub(from, amount).emit(encode_and_encrypt_note(
+            &mut context,
+            from,
+            from,
+        ));
+
+        // TODO(benesjan): Re-enable this check once #14432 is merged.
```

Comment:
#14432 is [the commitment in nullifier PR](https://github.com/AztecProtocol/aztec-packages/pull/14432)